x/lazyledgerapp/types/payformessage.go

```diff
@@ -0,0 +1,314 @@
+package types
+
+import (
+	"crypto/sha256"
+	"errors"
+	fmt "fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lazyledger/lazyledger-core/crypto/merkle"
+	core "github.com/lazyledger/lazyledger-core/types"
+	"github.com/lazyledger/nmt"
+)
+
+const (
+	TypeMsgPayforMessage                   = "payformessage"
+	TypeSignedTransactionDataPayForMessage = "signedtransactiondatapayformessage"
+	ShareSize                              = core.ShareSize
+	SquareSize                             = core.MaxSquareSize
+	NamespaceIDSize                        = core.NamespaceSize
+)
+
+///////////////////////////////////////
+// 	MsgWirePayForMessage
+///////////////////////////////////////
+
+var _ sdk.MsgRequest = &MsgWirePayForMessage{}
+
+func (msg *MsgWirePayForMessage) Route() string { return RouterKey }
+
+func (msg *MsgWirePayForMessage) Type() string { return TypeMsgPayforMessage }
+
+// ValidateBasic checks for valid namespace length, declared message size, share
+// commitments, signatures for those share commitments, and fulfills the sdk.Msg
+// interface
+func (msg *MsgWirePayForMessage) ValidateBasic() error {
+	pubK := msg.PubKey()
+
+	// ensure that the namespace id is of length == NamespaceIDSize
+	if len(msg.GetMessageNameSpaceId()) != NamespaceIDSize {
+		return fmt.Errorf(
+			"invalid namespace length: got %d wanted NamespaceIDSize",
+			len(msg.GetMessageNameSpaceId()),
```

Comment:
I think what John had in mind was rather:
```suggestion
			"invalid namespace length: got %d wanted %d",
			len(msg.GetMessageNameSpaceId()),
			NamespaceIDSize,
```